services/motion/builtin/builtin.go

```diff
@@ -282,14 +282,24 @@ func (ms *builtIn) MoveOnGlobe(
 		return false, errors.New("destination cannot be nil")
 	}
 
-	moveRequest, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
+	mr, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
 	if err != nil {
 		return false, err
 	}
 
+	maxReplans := -1
+	i := 0
+	if extra != nil {
```

Comment:
This isn't how we should be checking for if a map is empty as it will fail if the map is not nil, but happens to be empty:
```suggestion
	if len(extra) == 0 {
```

Reply:
ah, I see why this was ok before... never mind.